### PR TITLE
feat: bitcoin wallet utils

### PR DESCRIPTION
**Pull request by wainola**
## Description

Currently the examples need to use a lot of code to get signer for the two types of transactions. Since we already have some utils to fetch UTXO's, it could useful to have some minimal wallet utils that allows have more cleaner examples.

## Related Issue Or Context

Closes: #547 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation

## Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have ensured that all acceptance criteria (or expected behavior) from issue are met
- [ ] I have updated the documentation locally and in chainbridge-docs.
- [ ] I have added tests to cover my changes.
- [ ] I have ensured that all the checks are passing and green, I've signed the CLA bot

### CODE REVIEW

File: packages/utils/src/bitcoin/wallet.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+import type { PublicKeyParams } from './types.js';
+
+/**
+ * @category Bitcoin Wallet Helpers
+ * @description Return either tweakedSigner and publicKeyDropedDERHeader or derivedNode to sign a transaction
+ * @param {PublicKeyParams} - bip32, mnemonic, derivationPath, network, typeOfAddress
+ * @returns {Promise<{ tweakedSigner: Signer; publicKeyDropedDERHeader: Buffer } | { derivedNode: BIP32Interface }>}
+ */
+export const getPublicKey = async ({
```
Comment (Lykhoyda) [inline packages/utils/src/bitcoin/wallet.ts:16]: It would be nice to have a test for this util

File: packages/utils/src/bitcoin/wallet.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import { TypeOfAddress } from '@buildwithsygma/bitcoin';
```
Comment (saadahmsiddiqui) [inline packages/utils/src/bitcoin/wallet.ts:1]: @wainola I think of `utils` package as providing access access to `composite` utils which combine functionality of two or more SDK sub packages, like `evm` and `substrate` for `hasEnoughLiquidity`. Can we move `bitcoin` specific utils to bitcoin package itself? and only keep utils that share logic across packages?

**Review by saadahmsiddiqui — CHANGES_REQUESTED** [review]

File: packages/utils/src/bitcoin/wallet.ts (hunk 2 above)
Comment (wainola) [reply packages/utils/src/bitcoin/wallet.ts]: solved and move it to the btc folder

File: packages/bitcoin/src/utils/wallet.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+ * @returns {Promise<{ tweakedSigner: Signer; publicKeyDropedDERHeader: Buffer } | { derivedNode: BIP32Interface }>}
+ */
+export const getPublicKey = async ({
+  bip32,
+  mnemonic,
+  derivationPath,
+  network,
+  typeOfAddress,
+}: PublicKeyParams): Promise<
+  { tweakedSigner: Signer; publicKeyDropedDERHeader: Buffer } | { derivedNode: BIP32Interface }
```
Comment (saadahmsiddiqui) [inline packages/bitcoin/src/utils/wallet.ts:22]: define types in `types.ts`

**Comment by saadahmsiddiqui** [comment]
@wainola I think you need to update examples as well

File: examples/bitcoin-to-evm-fungible-transfer/src/transfer.p2tr.ts [hunk 4]
```diff
@@ -1,12 +1,11 @@
...
 } from "@buildwithsygma/bitcoin";
 import type { BitcoinTransferParams, UTXOData } from "@buildwithsygma/bitcoin";
 import { BIP32Factory } from "bip32";
-import { mnemonicToSeed } from "bip39";
-import { crypto, initEccLib, networks } from "bitcoinjs-lib";
-import { toXOnly } from "bitcoinjs-lib/src/psbt/bip371";
+import { initEccLib, networks, Signer } from "bitcoinjs-lib";
 import dotenv from "dotenv";
 import * as tinysecp from "tiny-secp256k1";
 import { broadcastTransaction, fetchUTXOS, getFeeEstimates, processUtxos } from '@buildwithsygma/utils'
```
Comment (saadahmsiddiqui) [inline examples/bitcoin-to-evm-fungible-transfer/src/transfer.p2tr.ts:11]: this should be

```typescript
import { broadcastTransaction, fetchUTXOS, getFeeEstimates, processUtxos } from '@buildwithsygma/utils'
```

**Review by Lykhoyda — APPROVED** [review]

**Pull request merged**